File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.java
class Solution {
    public int eraseOverlapIntervals(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> a[0]-b[0]);
        int count = 0;
        int l = 0, r = 1;
        while(r < intervals.length){
            // overlapping intervals
            if(intervals[r][0] < intervals[l][1]){
                if(intervals[l][1] > intervals[r][1]){
                    l = r;
                }
                r++;
                count++;
            }
            else{ // not overlapping intervals
                l = r;
                r++;
            }
        }
        
        return count;
    }
}

File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.js
/**
 * @param {number[][]} intervals
 * @return {number}
 */
var eraseOverlapIntervals = function(intervals) {
    
    intervals.sort((a, b) => a[0]-b[0]);
    
    let prevEnd = intervals[0][1];
    let count = 0;
    for(let idx = 1; idx < intervals.length; idx++){
        let [start, end] = intervals[idx];
        if(start >= prevEnd){
            prevEnd = end;
        }
        else{
            count++;
            prevEnd = Math.min(prevEnd, end);
        }
    }
    return count;
};